### PR TITLE
[CWS-1878] avoid syslog errors

### PR DESCRIPTION
**Pull request by arields1**

### CODE REVIEW

File: src/libbpf.c [hunk 1]
```diff
@@ -9196,6 +9196,37 @@ const char * libbpf_with_debugfs_location_prefix(const char * file)
...
+	static char path[sizeof(debugfs_customized_location) + 64] = {0};
+
+	if (name == NULL) {
+		return -1;
+	}
+
+	ret = snprintf(path, sizeof(path),
+					"%s/events/%s/s1_%s%s",
+					libbpf_get_debugfs_location(),
+					uprobe ? "uprobes" : "kprobes",
```
Comment (tomer-hadri-s1) [inline src/libbpf.c]: 1. for group, it should be `ebpf_group_prefix` (its not hooking to the default group as does s1-perf) and u can remove the `uprobe` arg
2. ident
3. try to test it manually, by adding expected entry to 'kprobe_events' and see if this logic is called or not (using `strace`)

File: src/libbpf.c [hunk 2]
```diff
@@ -9196,6 +9196,37 @@ const char * libbpf_with_debugfs_location_prefix(const char * file)
 	return path;
 }
 
+// Customization:
+static int probe_exsits(const char * name, bool uprobe, bool retprobe)
+{
+	char buf[STRERR_BUFSIZE];
+	int err, ret;
+	static char path[sizeof(debugfs_customized_location) + 64] = {0};
```
Comment (tomer-hadri-s1) [inline src/libbpf.c]: maybe put `128` instead of `64`? (just to be on the safe side)

File: src/libbpf.c [hunk 3]
```diff
@@ -9196,6 +9196,37 @@ const char * libbpf_with_debugfs_location_prefix(const char * file)
...
+		pr_warn("probe_exsits: failed: %s\n",
+				libbpf_strerror_r(err, buf, sizeof(buf)));
+		return ret;
+	}
+
+	if (exists(path)) {
+		return 1;
+	} else {
+		return 0;
+	}
```
Comment (tomer-hadri-s1) [inline src/libbpf.c]: consider:
1. just
`return exists(path);`
2. adjust return value type to `bool` (in lines 9207 + 9220, just `return false`)
3. and the condition in line 9277 will be clearer:
`if (!probe_exsits(isra_name_comp ? isra_name_comp : name, retprobe)) {` (note the `!`)

File: src/libbpf.c [hunk 4]
```diff
@@ -9243,6 +9274,10 @@ static int debugfs_clear_probe(
 
 	char * isra_name_comp = isra_symbol(name) ? make_isra_symbol_comp(name) : NULL;
 
+	if (probe_exsits(isra_name_comp ? isra_name_comp : name, uprobe, retprobe) == 0) {
```
Comment (tomer-hadri-s1) [inline src/libbpf.c]: missing `zfree(&isra_name_comp);`

File: src/libbpf.c [hunk 5]
```diff
@@ -9196,6 +9196,37 @@ const char * libbpf_with_debugfs_location_prefix(const char * file)
 	return path;
 }
 
+// Customization:
+static int probe_exsits(const char * name, bool uprobe, bool retprobe)
```
Comment (tomer-hadri-s1) [inline src/libbpf.c]: 1. typo: exsits -> exists
2. please name it with `debugfs_` prefix

**Review by tomer-hadri-s1 — COMMENTED** [review]
general:
1. regarding commit mesasge, it's dmesg warnings, no?
2. in addition, please add a short description (like: trying to remove non existing debugfs k[ret]probe hook etc)

File: src/libbpf.c (hunk 1 above)
Comment (arields1) [reply src/libbpf.c]: Done

File: src/libbpf.c (hunk 2 above)
Comment (arields1) [reply src/libbpf.c]: Done

File: src/libbpf.c (hunk 3 above)
Comment (arields1) [reply src/libbpf.c]: Done

File: src/libbpf.c (hunk 4 above)
Comment (arields1) [reply src/libbpf.c]: Done

File: src/libbpf.c (hunk 5 above)
Comment (arields1) [reply src/libbpf.c]: Done

**Review by tomer-hadri-s1 — APPROVED** [review]

**Pull request merged**